enhancements/security/openshift-image-policy.md

```diff
@@ -0,0 +1,359 @@
+
+## Test Plan
+
+Testing release signature validation is a gap in our current process, because we don't have triggers to run CI jobs after they've been signed _and_ a later release that we can update to has also been signed.
+We could fill this in with a periodic, but if we test once A and B have both been signed, it's likely that A has already been shipped, and it's too late to catch and fix any bugs the tests turn up in A.
+We could close this gap by signing CI releases with a CI key, [as envisioned years ago for OpenPGP signatures][ci-signing-key], but so far never implemented.
+It's unclear if this gap is considered severe enough to be worth that kind of investment, when occasional manual testing has done a fairly good job at keeping bugs out of slow-changing code (although we have had at least [one bug][verified-vs-graceful-shutdown] with signature handling that CI might have caught).
+
+## Graduation Criteria
+
+This functionality is expected to use [the existing `ImagePolicy` feature gate][feature-gate], and have the same graduation criteria.
+
+### Dev Preview -> Tech Preview
+
+[The existing `ImagePolicy` feature gate is already `TechPreviewNoUpgrade`][feature-gate], so this transition will never happen for this new functionality.
+
+### Tech Preview -> GA
+
+This functionality is expected to use [the existing `ImagePolicy` feature gate][feature-gate], and have the same graduation criteria.
+
+### Removing a deprecated feature
+
+This proposal does not deprecate any features.
+This proposal does not remove any features, with the possible exception of the removal of the OCP image scope guard discussed in [the *API Extensions* section](#api-extensions).
+
+## Upgrade / Downgrade Strategy
+
+Existing clusters can return to the previous behavior after updating by setting the `overrides` entry mentioned in [the *Installer* section](#installer) and removing the policy with:
+
+```console
+$ oc delete clusterimagepolicy openshift
+```
+
+## Version Skew Strategy
+
+The new ClusterImagePolicy manifest should have a late enough run-level that the cluster-version operator (CVO) applies it after the ClusterImagePolicy CRD and any machine-config tooling needed to process the custom resource.
+Because the machine-config is at run-level 80, late in the update, this means that updates into new image policies will be applied after most other components have rolled out their new images, which may require slower ratcheting for things like signing key rotations.
+
+## Operational Aspects of API Extensions
+
+None of the changes proposed in [the *API Extensions* section](#api-extensions) have operational impact.
+
+## Support Procedures
+
+I expect "images are failing to meet the relevant image policy/policies" to be something with a solid support experience, regardless of whether OpenShift defines its own image policy, or leaves image policies entirely up to users.
```

Comment:
I have no numbers, but I my guess is that this will be the _first_ time many users are going to encounter signed images.

Especially users with mirrors, and without a mirroring process set up to mirror signatures, could be surprised. (That’s currently mitigated by the feature gate on image policies.)

Reply:
> Especially users with mirrors, and without a mirroring process set up to mirror signatures...

We get a fair amount of questions about our current OpenPGP signatures in this situation.  And in fact, in [4.15 we built a `TechPreviewNoUpgrade` system to make that process easier][1].  We were [thinking of going GA in 4.17][2], but then this Sigstore push came swooping in, and now we may end up dropping the OpenPGP distribution framework.  Still up in the air.  Here][3] is a demo of the current OpenPGP verification failure experience.

But yes, I expect there will be folks with mirroring and signature-distribution hiccups that run into trouble.  We have those folks already with the current OpenPGP system.  I expect we'll still have them with any new system.  Distribution and mirroring is hard.  We want the "hey, your policy calls for a signature we can't find after checking..." experience to be reasonably solid.  But that applies to all ClusterImagePolicies and Pods.  I don't think it's unique to this enhancement's OpenShift-specific policy.

[1]: https://issues.redhat.com/browse/OTA-946
[2]: https://issues.redhat.com/browse/OTA-1118
[3]: https://github.com/openshift/cluster-version-operator/pull/1003#issuecomment-1848075607